Thème de coloration syntaxique accessible et aux couleurs de ZdS

AmauryCarrade opened the pull request:
Cette PR _propose_ un thème de coloration syntaxique basé sur le thème a11y-light, [que l'on peut retrouver par ici](https://highlightjs.org/static/demo/) (le sélectionner à gauche), mais adapté aux couleurs de ZdS.

L'objectif est de l'adapter également pour CodeMirror — ça ne devrait pas être trop compliqué — et éventuellement d'en faire un paquet npm externe, afin que d'autres projets de ZdS (par exemple, Zeste de Code) puissent en profiter.

### Aperçu

_N'hésitez pas à cliquer sur les images pour les ouvrir en bonne qualité._

_**Ces captures d'écran ne sont plus à jour.**_

![Lisp](https://user-images.githubusercontent.com/1417570/103250000-19a0cc80-4972-11eb-83bd-a7d105d3ea6d.png)

![Python](https://user-images.githubusercontent.com/1417570/103250008-29201580-4972-11eb-8adf-7fed1d7ac8e5.png)

![Rust](https://user-images.githubusercontent.com/1417570/103250021-32a97d80-4972-11eb-9808-0e33644fc83e.png)

![INI](https://user-images.githubusercontent.com/1417570/103250037-43f28a00-4972-11eb-8f9f-bb875abd0aed.png)

![C](https://user-images.githubusercontent.com/1417570/103250047-4f45b580-4972-11eb-9d62-808417f6b664.png)

### Contrôle qualité

1. Lancer le site avec `make run`.
2. Aller sur un contenu ou en créer un contenant du code source. On pourra utiliser [ce fichier Markdown contenant plein de code sources dans plein de langages](https://gist.githubusercontent.com/AmauryCarrade/c6459874f2e4981afaef7a546d983c9b/raw/376a19e686f2083a74275ad43ad27e521c8d2ded/que-de-couleurs.md).
3. Vérifier que le rendu est bon, lisible, et accessible.

AmauryCarrade committed:
Essai d'un thème de coloration ZdS

- Inspiré d'a11y-light
- Accessibilité vérifiée

AmauryCarrade committed:
Adaptation du thème pour CodeMirror

- Suppression du thème IDEA
- Intégration du thème au CSS principal

Situphen commented:
Tu peux enlever `highlight.js` de `package.json` étant donné qu'on ne s'en servira plus après cette PR.

AmauryCarrade commented:
Oh ? Bonne chose ça.

Enfin j'attends tout de même de savoir si ça vous plaît :)

AmauryCarrade committed:
Retrait d'une dépendance inutile

- On n'utilise plus highlight.js

AmauryCarrade committed:
Léger affinement de la colorisation du code

AmauryCarrade committed:
Le code glisse désormais

- La graisse de la police des codes sources a été augmentée, pour
  plus de lisibilité.

AmauryCarrade committed:
Amélioration continue de la coloration

AmauryCarrade committed:
Lignes plus discrètes en console

AmauryCarrade committed:
Lisibilité du CSS coloré

AmauryCarrade committed:
Amélioration de la coloration de CodeMirror

- Liens non soulignés
- Textes barrés mais pas leurs délimiteurs
- Titres complètement colorés et plus sombres

AmauryCarrade committed:
Les nombres sont plus visibles

AmauryCarrade committed:
Commentaires plus discrets

Situphen reviewed (approved):
**QA OK** Pour moi c'est bon !
